File: Solutions/1051. Height Checker/1051. Height Checker.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        int res=0,currh=0;
        vector<int> cnt(101,0);
        for(int h:heights){
            ++cnt[h];
        }
        for(int h:heights){
            while(!cnt[currh])
                ++currh;
            if(currh!=h)
                ++res;
            --cnt[currh];
        }
        return res;
    }
};

File: Solutions/1051. Height Checker/1051. Height Checker.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        res,currh=0,0
        cnt=[0]*101
        for h in heights:
            cnt[h]+=1
        for h in heights:
            while cnt[currh]==0:
                currh+=1
            if currh!=h:
                res+=1
            cnt[currh]-=1
        return res